R/utils.R

```diff
@@ -604,58 +604,84 @@ getColumnValues <- function(conn, tableName, columnName, where = NULL, mock = FA
       return (character(0))
   }else {
     sql = paste0("SELECT DISTINCT ", columnName, " values FROM ", tableName, where_clause)
-    
+
     rs = toaSqlQuery(conn, sql, stringsAsFactors=FALSE)
-    
+
     return (rs[, 'values'])
   }
 }
 
 #' Test if table present in the database.
-#' 
+#'
 #' @param channel object as returned by \code{\link{odbcConnect}}.
-#' @param names vector of table names. Name may contain SQL wildcard 
-#'   characters but it should not contain schema prefix. All visible schemas 
+#' @param names vector of table names. Name may contain SQL wildcard
+#'   characters but it should not contain schema prefix. All visible schemas
 #'   will be checked for table name specified.
 #' @return logical vector indicating if corresponding name is table in Aster database.
 #' @seealso \code{\link{getTableSummary}}
 #' @export
-#' @examples 
+#' @examples
 #' if(interactive()){
-#' # initialize connection to Lahman baseball database in Aster 
+#' # initialize connection to Lahman baseball database in Aster
 #' conn = odbcDriverConnect(connection="driver={Aster ODBC Driver};
 #'                          server=<dbhost>;port=2406;database=<dbname>;uid=<user>;pwd=<pw>")
-#' 
-#' isTable(conn, "pitching")        # TRUE 
+#'
+#' isTable(conn, "pitching")        # TRUE
 #' isTable(conn, "pitch%")          # TRUE
 #' isTable(conn, "public.pitching") # FALSE
 #' }
 isTable <- function(channel, names) {
   if (is.null(names) || length(names) < 1) return(logical(0))
-  
+
   vapply(names, FUN=function(x) nrow(sqlTables(channel, tableName=x))>0, FUN.VALUE=logical(1))
 }
 
 # TODO: make part of utility convinience set of functions
 grantExecuteOnFunction <- function(conn, name='%', owner='db_superuser', user) {
-  
+
   if (missing(user)) {
     stop("User name to grant execute permission is missing.")
   }
-  
+
   sql = paste0("select * from nc_system.nc_all_sqlmr_funcs where funcname like '", name, "' and funcowner = '",
                owner, "'")
-  
+
   func_list = toaSqlQuery(conn, sql)
-  
+
   if (!('funcname' %in% names(func_list)) || length(func_list$funcname) == 0) {
     stop("No functions to grant execute permission to.")
   }
-  
+
   for(func_name in func_list$funcname){
     sql = paste0("grant execute on function ", func_name, " to ", user)
     result = toaSqlQuery(conn, sql)
-    
+
     return (result)
   }
-}
\ No newline at end of file
+}
+
+
+## given character vector of package names
+## download packages and their dependencies from CRAN to a directory
+## and zip the dir
+getPkgDependenciesZip <- function(packageNames){
+    require(tools)
```

Comment:
neither 'require' nor 'library' don't belong to functions inside packages;
in this case we add packages in DESCRIPTION's Imports: section, and qualify functions from the packages with ::